client/java/src/main/java/io/openlineage/client/OpenLineageHttp.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.openlineage.client;
+
+import static org.apache.http.Consts.UTF_8;
+import static org.apache.http.HttpHeaders.ACCEPT;
+import static org.apache.http.HttpHeaders.CONTENT_TYPE;
+import static org.apache.http.entity.ContentType.APPLICATION_JSON;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.core.type.TypeReference;
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.Map;
+import javax.annotation.Nullable;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+
+@Slf4j
+class OpenLineageHttp implements Closeable {
+  private static final String BASE_PATH = "/api/v1";
+
+  final HttpClient http;
+  final URL url;
+  @Nullable final String apiKey;
+
+  OpenLineageHttp(
+      @NonNull final HttpClient http, @NonNull final URL url, @Nullable final String apiKey) {
+    this.http = http;
+    this.url = url;
+    this.apiKey = apiKey;
+  }
+
+  static OpenLineageHttp create(@NonNull final URL url, @Nullable final String apiKey) {
+    final CloseableHttpClient http = HttpClientBuilder.create().build();
+    return new OpenLineageHttp(http, url, apiKey);
+  }
+
+  String post(@NonNull URL url, @NonNull String json) {
+    log.debug("POST {}: {}", url, json);
+    try {
+      final HttpPost request = new HttpPost();
+      request.setURI(url.toURI());
```

Comment:
We're converting URIs to URLs as params to  `Clients` or `OpenLineageClient`, then convert to URI again.
We should just use URI class everywhere I think.

Especially since URL is evil 😉 
https://blog.cyberborean.org/2007/08/22/urls-in-java-another-pitfall